nixos/modules/services/networking/dns-over-https.nix

```diff
@@ -0,0 +1,193 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.dns-over-https;
+  clientBootstrap = concatMapStringsSep "," (x: "\"${x}\"") cfg.client.bootstrap;
+  clientListen = concatMapStringsSep "," (x: "\"${x}\"") cfg.client.listen;
+  clientUpstream = concatMapStrings (s: "[[upstream.upstream_ietf]]\n  url = \"${s.url}\"\n  weight = ${toString s.weight}\n") cfg.client.upstream;
+  serverListen = concatMapStringsSep "," (x: "\"${x}\"") cfg.server.listen;
+  serverUpstream = concatMapStringsSep "," (x: "\"${x}\"") cfg.server.upstream;
+
+  # See upstream: https://github.com/m13253/dns-over-https/blob/master/doh-client/doh-client.conf
+  clientConf = pkgs.writeText "doh-client.conf" ''
+    listen = [${clientListen}]
+
+    [upstream]
+    upstream_selector = "${cfg.client.upstreamSelector}"
+
+    ${clientUpstream}
+
+    [others]
+    bootstrap = [${clientBootstrap}]
+    passthrough = [
+        "captive.apple.com",
+        "connectivitycheck.gstatic.com",
+        "detectportal.firefox.com",
+        "msftconnecttest.com",
+        "nmcheck.gnome.org",
+        "pool.ntp.org",
+        "time.apple.com",
+        "time.asia.apple.com",
+        "time.euro.apple.com",
+        "time.nist.gov",
+        "time.windows.com",
```

Comment:
Would you want to expand those? E.g. ubuntu.pool.ntp.org.

Reply:
Made it an option